### ISSUE TITLE
There might be a small bug in optimize_prior_precision

### ISSUE THREAD

**Issue by ruili-pml** [aleximmer/Laplace#248]
Hi,

There might be a small bug in https://github.com/aleximmer/Laplace/blob/6b0618aed74cb305f31a114fab81774976ed96f6/laplace/utils/metrics.py#L35
as for classification, it requires the target to be the label, instead of one hot vector.  

Since in the train_loader it requires one hot vector (or maybe the label also work, in my case I only feed into the one hot vector), it might be consistent to let the val_loader require one hot vector or make both format work?

Best,
Rui
